python/pyspark/pandas/utils.py

```diff
@@ -600,7 +600,7 @@ def column_labels_level(column_labels: List[Tuple]) -> int:
         return list(levels)[0]
 
 
-def name_like_string(name: Optional[Union[str, Tuple]]) -> str:
+def name_like_string(name: Optional[Union[Any, Tuple]]) -> str:
```

Comment:
Question: Is this change required?

Reply:
Good point!
Actually `str` here was a mistake because the `name` in pandas-on-Spark can be non-string values, like `int`, `float`, etc., and `mypy` reported an error.